ui/app/contexts/metametrics.new.js

```diff
@@ -0,0 +1,180 @@
+
+export const MetaMetricsContext = createContext(() => {
+  captureException(
+    Error(`MetaMetrics context function was called from a react node that is not a descendant of a MetaMetrics context provider`),
+  )
+})
+
+const PATHS_TO_CHECK = Object.keys(PATH_NAME_MAP)
+
+export function MetaMetricsProvider ({ children }) {
+  const network = useSelector(getCurrentNetworkId)
+  const metaMetricsId = useSelector((state) => state.metamask.metaMetricsId)
+  const participateInMetaMetrics = useSelector((state) => state.metamask.participateInMetaMetrics)
+  const metaMetricsSendCount = useSelector((state) => state.metamask.metaMetricsSendCount)
+  const location = useLocation()
+
+  /**
+   * Anytime the location changes, track a page change with segment.
+   * Previously we would manually track changes to history and keep a
+   * reference to the previous url, but with page tracking we can see
+   * which page the user is on and their navigation path.
+   */
+  useEffect(() => {
+    if (location.pathname.startsWith('/initialize') || participateInMetaMetrics) {
+      // Events that happen during initialization before the user opts into MetaMetrics will be anonymous
+      const idTrait = metaMetricsId ? 'userId' : 'anonymousId'
+      const idValue = metaMetricsId ?? METAMETRICS_ANONYMOUS_ID
+      const match = matchPath(location.pathname, { path: PATHS_TO_CHECK, exact: true, strict: true })
+      if (match) {
+        const { path, params, url } = match
+        const name = PATH_NAME_MAP[path]
+        segment.page({
+          [idTrait]: idValue,
+          name,
+          properties: {
+            url,
+            hash: location.hash,
+            // We do not want to send addresses or accounts in any events
+            // Some routes include these as params.
+            params: omit(params, ['account', 'address']),
+          },
+        })
+      } else if (location.pathname !== '/confirm-transaction') {
+        // We have more specific pages for each type of transaction confirmation
+        // The user lands on /confirm-transaction first, then is redirected based on
+        // the contents of state.
+        captureMessage(`${location.pathname} would have issued a page track event to segment, but no route match was found`)
+      }
+    }
+  }, [location, metaMetricsId, participateInMetaMetrics])
+
+  /**
+   * track a metametrics event using segment
+   * e.g metricsEvent({ event: 'Unlocked MetaMask', category: 'Navigation' })
+   *
+   * @param {object}   config - configuration object for the event to track
+   * @param {string}   config.event - event name to track
+   * @param {string}   config.category - category to associate event to
+   * @param {boolean}  [config.isOptIn] - happened during opt in/out workflow
+   * @param {object}   [config.properties] - object of custom values to track, snake_case
+   * @return {undefined}
+   */
+  const trackEvent = useCallback(
+    (config = {}) => {
+      console.log(config)
```

Comment:
Nit: leftover `console.log`

We should consider adding this to the lint rules :thinking:

Reply:
removed.